### PR TITLE
Deploy fluidity gauge reward forwarder

### PR DESCRIPTION
**Pull request by mjdatadog**
Deploys
- Root/Child Gauge
- Reward Forwarder for child gauge
- Reward Scheduler for child gauge

- Adds reward token to child gauge

Updates deployUtils to check for existence of gauges so can be run to check sanity of anycall deploy.

Intended Deploy process:
1. Deploy Root Gauge on mainnet
2. Run arbitrum deployments
3. Call add_gauge from the multisig to add root gauge to controller

### CODE REVIEW

File: deploy/arbitrum/261_depoy_fluidity_RewardScheduler.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  const { deployer } = await getNamedAccounts()
+
+  const RewardScheduler = await deploy(
+    "RewardScheduler_ChildGauge_CommunityfUSDCLPToken",
+    {
+      from: deployer,
+      contract: "RewardScheduler",
+      log: true,
+      skipIfAlreadyDeployed: true,
+      args: [fUSDC_Reward_Manager_Address, (await get("fUSDC")).address],
```
Comment (penandlim) [inline deploy/arbitrum/261_depoy_fluidity_RewardScheduler.ts:18]: Looks like a wrong address used for RewardForwarder associated with the ChildGauge of CommunityfUSDCPoolLPToken. Should be the RewardForwarder.

File: deploy/arbitrum/261_depoy_fluidity_RewardScheduler.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    },
+  )
+
+  // Transfer ownership to fUSDC_Reward_Manager_Address
+  const rewardScheduler = await ethers.getContractAt(
+    "RewardScheduler",
+    RewardScheduler.address,
+  )
+  await rewardScheduler.transferOwnership(fUSDC_Reward_Manager_Address)
```
Comment (penandlim) [inline deploy/arbitrum/261_depoy_fluidity_RewardScheduler.ts:27]: ethers.js library should be avoided in deploy scripts. The output logs do not include output from this.
Instead hardhat-deploy style tx execution should be used.

**Review by penandlim — CHANGES_REQUESTED** [review]

File: deploy/arbitrum/250_deploy_child_gauges.ts [hunk 3]
```diff
@@ -16,6 +16,7 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
     SaddleFRAXUSDTMetaPoolLPToken: "FRAXBP-USDT",
     SaddleFRAXUSDsMetaPoolLPToken: "FRAXBP-USDs",
     SaddleUSXFRAXBPMetaPoolLPToken: "USX-FRAXBP",
+    CommunityfUSDCPoolLPToken: "fUSDC-USDC",
```
Comment (penandlim) [inline deploy/arbitrum/250_deploy_child_gauges.ts:19]: Registry mismatch will lead to mismatched deployment address of Root/Child. Should use same name mapping on both mainnet and sidechain.

**Review by penandlim — CHANGES_REQUESTED** [review]

**Review by penandlim — APPROVED** [review]
LGTM

**Comment by penandlim** [comment]
Both gauges are deployed at 0xC7ec37B1E3be755e06a729e11a76ff4259768F12
RewardScheduler https://arbiscan.io/address/0x9d7fccb3085f87298c365a59df9084f76fb93c0b#code
RewardForwarder https://arbiscan.io/address/0xf428d446670d901ad6893fe2dbc77d42fcd20cb2#code

**Comment by penandlim** [comment]
TODO: Add the RewardForwarder and the reward token to the ChildGauge to enable the fUSDC reward for the gauge.

**Review by penandlim — DISMISSED** [review]

File: deploy/arbitrum/261_depoy_fluidity_RewardScheduler.ts [hunk 4]
```diff
@@ -26,11 +25,12 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
-    "transferOwnership",
-    fUSDC_Reward_Manager_Address,
-  )
+  // Commenting out for forked deployments
+  // const rewardSchedulerOwnershipTX = await execute(
+  //   "RewardScheduler_ChildGauge_CommunityfUSDCLPToken",
+  //   { from: deployer, log: true },
+  //   "transferOwnership",
+  //   fUSDC_Reward_Manager_Address,
+  // )
```
Comment (penandlim) [inline deploy/arbitrum/261_depoy_fluidity_RewardScheduler.ts:34]: We should never comment out code that may be required to recreate the state.

Instead, ideally, we should add checks for skipping this call based on other factors such that the script will function as expected in more cases.

**Review by penandlim — DISMISSED** [review]

File: test/xchainGauges/childGauge.ts [hunk 5]
```diff
@@ -347,6 +354,24 @@ describe("ChildGauge", () => {
...
+        // 100 tokens distributed over 1 week, 1 day passed
+        // 100 * 1e18 / 7
+        await expect(
+          childGauge.connect(signers[1])["claim_rewards()"](),
+        ).to.changeTokenBalance(
+          dummyRewardToken,
+          users[1],
+          "14285879629629599900",
+        )
+      })
```
Comment (penandlim) [inline test/xchainGauges/childGauge.ts:374]: Hmm I believe the setup for the edge case we want to test against is:
`add_reward(randomEOA, rewardForwarder)`

Case where we use an EOA address instead of an ERC20 address for the token. The main reason for this test is to see if the ChildGauge contract would make any calls to `randomEOA` that would fail due to the fact that it is NOT erc20 or a contract at all.

Although after manually looking into ChildGauge, I think it wouldn't make any call to `randomEOA` since `total_claimable` will always be 0, we should try to confirm this via test.

https://github.com/saddle-finance/saddle-contract/blob/e7b2258152df959ce26c85e03a6cd3f34f2a0577/contracts/xchainGauges/implementations/ChildGauge.vy#L226-L252

**Review by penandlim — APPROVED** [review]

**Review by penandlim — APPROVED** [review]
